catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resourceretriever/LocalResourceRetriever.java

```diff
@@ -61,44 +73,113 @@ public ResourceResponse retrieveResource(long bytesToSkip) throws ResourceNotFou
-        if (reader.getSupportedSchemes().contains(scheme)) {
-          try {
+      if (reader != null && reader.getSupportedSchemes().contains(scheme)) {
+        try {
+          LOGGER.debug(
+              "Found an acceptable resource reader ({}) for URI {}",
+              reader.getId(),
+              resourceRetrievalUri.toASCIIString());
+          resource = reader.retrieveResource(resourceRetrievalUri, props);
+          if (resource != null) {
+            break;
+          } else {
             LOGGER.debug(
-                "Found an acceptable resource reader ({}) for URI {}",
+                "Resource returned from ResourceReader {} was null. Checking other readers for URI: {}",
                 reader.getId(),
-                resourceUri.toASCIIString());
-            resource = reader.retrieveResource(resourceUri, props);
-            if (resource != null) {
-              break;
-            } else {
-              LOGGER.debug(
-                  "Resource returned from ResourceReader {} was null. Checking other readers for URI: {}",
-                  reader.getId(),
-                  resourceUri);
-            }
-          } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
-            LOGGER.debug("Product not found using resource reader with name {}", reader.getId(), e);
+                resourceRetrievalUri);
           }
+        } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
+          LOGGER.debug("Product not found using resource reader with name {}", reader.getId(), e);
         }
       }
     }
 
     if (resource == null) {
       throw new ResourceNotFoundException(
           "Resource Readers could not find resource (or returned null resource) for URI: "
-              + resourceUri.toASCIIString()
+              + resourceRetrievalUri.toASCIIString()
               + ". Scheme: "
-              + resourceUri.getScheme());
+              + resourceRetrievalUri.getScheme());
     }
     LOGGER.debug("Received resource, sending back: {}", resource.getResource().getName());
     LOGGER.trace("EXITING: {}", methodName);
 
     return resource;
   }
+
+  private URI getDerivedUriWithFragment(Metacard metacard, String fragment) {
+    if (!attributeIsPresent(metacard, Core.DERIVED_RESOURCE_URI)) {
+      return null;
+    }
+
+    Attribute attribute = metacard.getAttribute(Core.DERIVED_RESOURCE_URI);
+
+    List<Serializable> derivedUris = attribute.getValues();
+    if (!derivedUris.isEmpty()) {
+      for (Serializable uri : derivedUris) {
+        if (!(uri instanceof String)) {
+          continue;
+        }
+
+        URI derivedUri = null;
+        try {
+          derivedUri = new URI((String) uri);
+        } catch (URISyntaxException e) {
+          LOGGER.debug(
+              "Received invalid [{}] [{}] on metacard with id [{}].",
+              Core.DERIVED_RESOURCE_URI,
+              uri,
+              metacard.getId());
```

Comment:
If you continue here you could just do a `derivedUri.getFragment().equals(fragment)` in the if below